<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Part 4 A toy bioinformatics project | Introduction to Linux for the HPC</title>
  <meta name="description" content="A book for the Introduction to Linux for the HPC workshop." />
  <meta name="generator" content="bookdown 0.29 and GitBook 2.6.7" />

  <meta property="og:title" content="Part 4 A toy bioinformatics project | Introduction to Linux for the HPC" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A book for the Introduction to Linux for the HPC workshop." />
  <meta name="github-repo" content="vari-bbc/Intro_to_Linux_for_HPC" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Part 4 A toy bioinformatics project | Introduction to Linux for the HPC" />
  
  <meta name="twitter:description" content="A book for the Introduction to Linux for the HPC workshop." />
  

<meta name="author" content="BBC" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="hpc-basics-and-the-job-scheduler.html"/>

<script src="assets/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="assets/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<script src="assets/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="assets/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="assets/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="assets/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#workshop-organizers"><i class="fa fa-check"></i><b>1.1</b> Workshop organizers</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="basics-of-using-the-hpc-and-the-linux-commandline.html"><a href="basics-of-using-the-hpc-and-the-linux-commandline.html"><i class="fa fa-check"></i><b>2</b> Basics of using the HPC and the Linux commandline</a><ul>
<li class="chapter" data-level="2.1" data-path="basics-of-using-the-hpc-and-the-linux-commandline.html"><a href="basics-of-using-the-hpc-and-the-linux-commandline.html#ex-1-log-into-the-hpc"><i class="fa fa-check"></i><b>2.1</b> Ex 1 Log into the HPC</a></li>
<li class="chapter" data-level="2.2" data-path="basics-of-using-the-hpc-and-the-linux-commandline.html"><a href="basics-of-using-the-hpc-and-the-linux-commandline.html#commands-you-need-to-learn-for-ex2"><i class="fa fa-check"></i><b>2.2</b> Commands you need to learn for Ex2:</a></li>
<li class="chapter" data-level="2.3" data-path="basics-of-using-the-hpc-and-the-linux-commandline.html"><a href="basics-of-using-the-hpc-and-the-linux-commandline.html#ex2-file-navigation"><i class="fa fa-check"></i><b>2.3</b> Ex2: file navigation</a></li>
<li class="chapter" data-level="2.4" data-path="basics-of-using-the-hpc-and-the-linux-commandline.html"><a href="basics-of-using-the-hpc-and-the-linux-commandline.html#commands-you-need-to-learn-for-ex-3"><i class="fa fa-check"></i><b>2.4</b> Commands you need to learn for Ex 3</a><ul>
<li class="chapter" data-level="2.4.1" data-path="basics-of-using-the-hpc-and-the-linux-commandline.html"><a href="basics-of-using-the-hpc-and-the-linux-commandline.html#so-what-is-pipe"><i class="fa fa-check"></i><b>2.4.1</b> So, what is pipe?</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="basics-of-using-the-hpc-and-the-linux-commandline.html"><a href="basics-of-using-the-hpc-and-the-linux-commandline.html#exercise-3-file-viewing-and-manipulation"><i class="fa fa-check"></i><b>2.5</b> Exercise 3: File viewing and manipulation</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="hpc-basics-and-the-job-scheduler.html"><a href="hpc-basics-and-the-job-scheduler.html"><i class="fa fa-check"></i><b>3</b> HPC basics and the job scheduler</a></li>
<li class="chapter" data-level="4" data-path="a-toy-bioinformatics-project.html"><a href="a-toy-bioinformatics-project.html"><i class="fa fa-check"></i><b>4</b> A toy bioinformatics project</a><ul>
<li class="chapter" data-level="4.1" data-path="a-toy-bioinformatics-project.html"><a href="a-toy-bioinformatics-project.html#start-up-an-interactive-job"><i class="fa fa-check"></i><b>4.1</b> Start up an interactive job</a></li>
<li class="chapter" data-level="4.2" data-path="a-toy-bioinformatics-project.html"><a href="a-toy-bioinformatics-project.html#create-a-project-directory-for-yourself-and-a-subdirectory-for-storing-the-raw-fastq-files"><i class="fa fa-check"></i><b>4.2</b> Create a project directory for yourself and a subdirectory for storing the raw fastq files</a></li>
<li class="chapter" data-level="4.3" data-path="a-toy-bioinformatics-project.html"><a href="a-toy-bioinformatics-project.html#copy-fastqs-to-working-directory"><i class="fa fa-check"></i><b>4.3</b> Copy fastqs to working directory</a></li>
<li class="chapter" data-level="4.4" data-path="a-toy-bioinformatics-project.html"><a href="a-toy-bioinformatics-project.html#check-that-the-files-transferred-properly"><i class="fa fa-check"></i><b>4.4</b> Check that the files transferred properly</a></li>
<li class="chapter" data-level="4.5" data-path="a-toy-bioinformatics-project.html"><a href="a-toy-bioinformatics-project.html#use-zcat-to-take-a-look-into-fastq.gz-files"><i class="fa fa-check"></i><b>4.5</b> Use zcat to take a look into fastq.gz files</a></li>
<li class="chapter" data-level="4.6" data-path="a-toy-bioinformatics-project.html"><a href="a-toy-bioinformatics-project.html#use-wc-to-see-how-many-reads-are-in-a-fastq-file-and-how-long-they-are."><i class="fa fa-check"></i><b>4.6</b> Use <code>wc</code> to see how many reads are in a fastq file and how long they are.</a></li>
<li class="chapter" data-level="4.7" data-path="a-toy-bioinformatics-project.html"><a href="a-toy-bioinformatics-project.html#how-to-see-what-packages-are-installed-on-hpc."><i class="fa fa-check"></i><b>4.7</b> How to see what packages are installed on HPC.</a></li>
<li class="chapter" data-level="4.8" data-path="a-toy-bioinformatics-project.html"><a href="a-toy-bioinformatics-project.html#run-fastqc-on-the-fastq-files"><i class="fa fa-check"></i><b>4.8</b> Run FastQC on the fastq files</a></li>
<li class="chapter" data-level="4.9" data-path="a-toy-bioinformatics-project.html"><a href="a-toy-bioinformatics-project.html#set-up-a-job-to-run-salmon"><i class="fa fa-check"></i><b>4.9</b> Set up a job to run Salmon</a></li>
<li class="chapter" data-level="4.10" data-path="a-toy-bioinformatics-project.html"><a href="a-toy-bioinformatics-project.html#submit-the-job"><i class="fa fa-check"></i><b>4.10</b> Submit the job</a></li>
<li class="chapter" data-level="4.11" data-path="a-toy-bioinformatics-project.html"><a href="a-toy-bioinformatics-project.html#check-the-job-logs-to-see-if-job-finished-running"><i class="fa fa-check"></i><b>4.11</b> Check the job logs to see if job finished running</a></li>
<li class="chapter" data-level="4.12" data-path="a-toy-bioinformatics-project.html"><a href="a-toy-bioinformatics-project.html#use-grep-to-find-the-tpms-for-specific-genes"><i class="fa fa-check"></i><b>4.12</b> Use grep to find the TPMs for specific genes</a></li>
<li class="chapter" data-level="4.13" data-path="a-toy-bioinformatics-project.html"><a href="a-toy-bioinformatics-project.html#use-an-interactive-job-to-run-multiqc-on-the-salmon-and-fastqc-output"><i class="fa fa-check"></i><b>4.13</b> Use an interactive job to run multiQC on the Salmon and FastQC output</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to Linux for the HPC</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="a-toy-bioinformatics-project" class="section level1 hasAnchor">
<h1><span class="header-section-number">Part 4</span> A toy bioinformatics project<a href="a-toy-bioinformatics-project.html#a-toy-bioinformatics-project" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Here, we will put together some of the new skills that we have discussed today to perform a toy bioinformatic analysis. Four RNA-seq fastq files representing paired-end reads from two human samples have already been downloaded from <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE52778">GSE52778</a>. To save time for the workshop, each file has been randomly subsetted to just a small fraction of the total reads. Here we will:</p>
<ol style="list-style-type: decimal">
<li>Copy these fastq files to a private project directory for each user.</li>
<li>Use basic Linux commands to learn some characteristics of these files, and compare the results to results from FastQC, which we will also run.</li>
<li>Use Salmon to align (pseudo-align) these reads to the hg38 reference transcriptome and get transcripts-per-million (TPMs) for each annotated gene.</li>
</ol>
<div id="start-up-an-interactive-job" class="section level2 hasAnchor">
<h2><span class="header-section-number">4.1</span> Start up an interactive job<a href="a-toy-bioinformatics-project.html#start-up-an-interactive-job" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Remember we try to avoid running computationally-intensive operations on the submit node directly. Where a formal script is not needed, we can start up an ‘interactive job’. Once an interactive job is started, we can use the commandline as before but now we are using pre-allocated computational resources.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="a-toy-bioinformatics-project.html#cb1-1"></a><span class="ex">qsub</span> -I -l nodes=1:ppn=1 -l mem=32gb -l walltime=2:00:00</span>
<span id="cb1-2"><a href="a-toy-bioinformatics-project.html#cb1-2"></a></span>
<span id="cb1-3"><a href="a-toy-bioinformatics-project.html#cb1-3"></a><span class="co"># one way to verify that we have started an interactive successfully is to run qstat</span></span>
<span id="cb1-4"><a href="a-toy-bioinformatics-project.html#cb1-4"></a><span class="co"># to show only jobs submitted by ourselves, we use the `-u` option followed by our username.</span></span>
<span id="cb1-5"><a href="a-toy-bioinformatics-project.html#cb1-5"></a><span class="ex">qstat</span> -u username</span></code></pre></div>
</div>
<div id="create-a-project-directory-for-yourself-and-a-subdirectory-for-storing-the-raw-fastq-files" class="section level2 hasAnchor">
<h2><span class="header-section-number">4.2</span> Create a project directory for yourself and a subdirectory for storing the raw fastq files<a href="a-toy-bioinformatics-project.html#create-a-project-directory-for-yourself-and-a-subdirectory-for-storing-the-raw-fastq-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>It is good practice to try to make a separate directory for each project to minimize the risk of accidentally editing other files. It is also good to store raw data in its own location and not make any modifications to them.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="a-toy-bioinformatics-project.html#cb2-1"></a><span class="bu">cd</span> /varidata/researchtemp/hpctmp/HPC_mini_workshop/Part3</span>
<span id="cb2-2"><a href="a-toy-bioinformatics-project.html#cb2-2"></a></span>
<span id="cb2-3"><a href="a-toy-bioinformatics-project.html#cb2-3"></a><span class="fu">mkdir</span> firstname.lastname</span>
<span id="cb2-4"><a href="a-toy-bioinformatics-project.html#cb2-4"></a></span>
<span id="cb2-5"><a href="a-toy-bioinformatics-project.html#cb2-5"></a><span class="bu">cd</span> firstname.lastname</span>
<span id="cb2-6"><a href="a-toy-bioinformatics-project.html#cb2-6"></a></span>
<span id="cb2-7"><a href="a-toy-bioinformatics-project.html#cb2-7"></a><span class="fu">mkdir</span> fastqs</span></code></pre></div>
</div>
<div id="copy-fastqs-to-working-directory" class="section level2 hasAnchor">
<h2><span class="header-section-number">4.3</span> Copy fastqs to working directory<a href="a-toy-bioinformatics-project.html#copy-fastqs-to-working-directory" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="a-toy-bioinformatics-project.html#cb3-1"></a><span class="co"># copy the md5sum file which will be used to verify successful file transfers.</span></span>
<span id="cb3-2"><a href="a-toy-bioinformatics-project.html#cb3-2"></a><span class="fu">cp</span> /varidata/researchtemp/hpctmp/BBC/hpc_workshop_fqs/md5sum.txt ./fastqs/</span>
<span id="cb3-3"><a href="a-toy-bioinformatics-project.html#cb3-3"></a></span>
<span id="cb3-4"><a href="a-toy-bioinformatics-project.html#cb3-4"></a><span class="co"># copy the fastqs</span></span>
<span id="cb3-5"><a href="a-toy-bioinformatics-project.html#cb3-5"></a><span class="fu">cp</span> /varidata/researchtemp/hpctmp/BBC/hpc_workshop_fqs/*fastq.gz ./fastqs/</span></code></pre></div>
</div>
<div id="check-that-the-files-transferred-properly" class="section level2 hasAnchor">
<h2><span class="header-section-number">4.4</span> Check that the files transferred properly<a href="a-toy-bioinformatics-project.html#check-that-the-files-transferred-properly" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Run the following command. The tool will compare the md5sums in the <code>md5sum.txt</code> file to the md5sums calculated for the files in your directory currently. If each line says ‘OK’, then the transfer was successful.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="a-toy-bioinformatics-project.html#cb4-1"></a><span class="bu">cd</span> fastqs</span>
<span id="cb4-2"><a href="a-toy-bioinformatics-project.html#cb4-2"></a></span>
<span id="cb4-3"><a href="a-toy-bioinformatics-project.html#cb4-3"></a><span class="ex">md5sum</span> -c md5sum.txt</span></code></pre></div>
<pre><code>## SRR1039520_1.fastq.gz: OK
## SRR1039520_2.fastq.gz: OK
## SRR1039521_1.fastq.gz: OK
## SRR1039521_2.fastq.gz: OK</code></pre>
<p>For the next steps, go back to the project directory.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="a-toy-bioinformatics-project.html#cb6-1"></a><span class="bu">cd</span> ..</span></code></pre></div>
</div>
<div id="use-zcat-to-take-a-look-into-fastq.gz-files" class="section level2 hasAnchor">
<h2><span class="header-section-number">4.5</span> Use zcat to take a look into fastq.gz files<a href="a-toy-bioinformatics-project.html#use-zcat-to-take-a-look-into-fastq.gz-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Note that a <code>.gz</code> file suffix means that the file is compressed and running the usual <code>cat</code> on it will not return human-readable results. The <code>zcat</code> command first decompresses the file then prints the results.</p>
<p>Note the matching fastq IDs in the R1 and R2 files. Valid paired fastq files always have matching read IDs throughout the entire file.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="a-toy-bioinformatics-project.html#cb7-1"></a><span class="fu">zcat</span> fastqs/SRR1039520_1.fastq.gz <span class="kw">|</span> <span class="fu">head</span></span></code></pre></div>
<pre><code>## @SRR1039520.19151550 19151550 length=63
## CCAGGACATCAAGAAGCCAGCTGAAGATGAGTGGGGTAAAACCCCAGACGCCATGAAAGCTGC
## +
## HJJJJJJJJJJJJJJJIJJJJJJJJIJJJJJGGJJJFHIJIIJJIGHHFFFDDDDDCDDDCDD
## @SRR1039520.3404519 3404519 length=63
## TGAGACATGGTTATAGATAAGAGAGTACAAAATGACTCTTTTTCCTGTCAATTGAAATTTAAA
## +
## HIGDFBGIIJBHGIIEHIIJIJIIIEGGIIIHGIJJIJJJIIJGIIIIGIEHIGDCHHIICG7
## @SRR1039520.16253787 16253787 length=63
## CAGGAGACCAAAGACACTGCAATTTGTGTGTTTTCTACAGGGTGCTTTAGATGACGTCTCATT</code></pre>
<p>Now look at the R2 file.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="a-toy-bioinformatics-project.html#cb9-1"></a><span class="fu">zcat</span> fastqs/SRR1039520_2.fastq.gz <span class="kw">|</span> <span class="fu">head</span></span></code></pre></div>
<pre><code>## @SRR1039520.19151550 19151550 length=63
## GAGATGGGGGTCCGTGCGGGCAGAACCCAGGGCATGAAGATCCAAAAGGGCCTGGTTCAGCTT
## +
## HJJJJJJJJJHHHIGIJIJJJJJJJHHHHFFFDDEEDDDDDDDDDDDDDDDDDDDDDDDDDDD
## @SRR1039520.3404519 3404519 length=63
## TTTGACCCTAGTATTGGCAATAGCCCTTTGCTATTTATATAATTAAAACTTTTCTTTAAATTT
## +
## HIJIJJJJIIEHGIGIIIDGIJJJJIIJJJIJJJIJJJIIIJJJJIJJGCHIJJJJJIJIJII
## @SRR1039520.16253787 16253787 length=63
## TACAGTTTGCAAAAGATGTCCAGATGGGTTCTTCTCAAATGAGACGTCATCTAAAGCACCCTG</code></pre>
</div>
<div id="use-wc-to-see-how-many-reads-are-in-a-fastq-file-and-how-long-they-are." class="section level2 hasAnchor">
<h2><span class="header-section-number">4.6</span> Use <code>wc</code> to see how many reads are in a fastq file and how long they are.<a href="a-toy-bioinformatics-project.html#use-wc-to-see-how-many-reads-are-in-a-fastq-file-and-how-long-they-are." class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Remember that we can use <code>man wc</code> to see what the different options such as <code>-l</code> and <code>-m</code> do.</p>
<p>This command counts the new line characters at the end of each line. Typically, this corresponds to the number of lines in a file. Recall that each read is represented by 4 lines in a fastq file, so we need to divide the results of <code>wc -l</code> by 4 to get the number of reads.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb11-1"><a href="a-toy-bioinformatics-project.html#cb11-1"></a><span class="fu">zcat</span> fastqs/SRR1039520_1.fastq.gz <span class="kw">|</span> <span class="fu">wc</span> -l</span></code></pre></div>
<pre><code>## 2000000</code></pre>
<p>This command counts the number of characters in the sequence line (second line) in the first read in the file. you may notice that the results of <code>wc -m</code> will be 1 higher than the actual read length. This is because the tool counts the newline character also.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb13-1"><a href="a-toy-bioinformatics-project.html#cb13-1"></a><span class="fu">zcat</span> fastqs/SRR1039520_1.fastq.gz <span class="kw">|</span> <span class="fu">head</span> -n2 <span class="kw">|</span> <span class="fu">tail</span> -n1 <span class="kw">|</span> <span class="fu">wc</span> -m</span></code></pre></div>
<pre><code>## 64</code></pre>
<p>Record these numbers so that you can compare these to the output from FastQC (a commonly used tool for checking the quality of fastq files) which we will run below.</p>
</div>
<div id="how-to-see-what-packages-are-installed-on-hpc." class="section level2 hasAnchor">
<h2><span class="header-section-number">4.7</span> How to see what packages are installed on HPC.<a href="a-toy-bioinformatics-project.html#how-to-see-what-packages-are-installed-on-hpc." class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>If you simply type <code>module av</code>, you will see all the modules available. There are a lot of modules installed. To parse through these more easily, we can <strong>pipe</strong> the results of <code>module av</code> into tools such as <code>head</code> to print just the first ‘n’ lines and <code>grep</code> to search for modules with specific keywords. There is a trick, though. The results of <code>module av</code> go to stderr, so we need to redirect stderr to stdout using <code>2&gt;&amp;1</code> before the pipe.</p>
<p>Print the first 50 lines.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="a-toy-bioinformatics-project.html#cb15-1"></a><span class="ex">module</span> av <span class="op">2&gt;&amp;1</span> <span class="kw">|</span> <span class="fu">head</span> -n50</span></code></pre></div>
<pre><code>## 
## ---------------------------- /cm/local/modulefiles -----------------------------
## cluster-tools/7.3 freeipmi/1.5.2    module-git        openldap
## cmd               gcc/6.1.0         module-info       shared
## dot               ipmitool/1.8.17   null              singularity/2.4.2
## 
## ---------------------------- /cm/shared/modulefiles ----------------------------
## acml/gcc/64/5.3.1
## acml/gcc/fma4/5.3.1
## acml/gcc/mp/64/5.3.1
## acml/gcc/mp/fma4/5.3.1
## acml/gcc-int64/64/5.3.1
## acml/gcc-int64/fma4/5.3.1
## acml/gcc-int64/mp/64/5.3.1
## acml/gcc-int64/mp/fma4/5.3.1
## acml/open64/64/5.3.1
## acml/open64/fma4/5.3.1
## acml/open64/mp/64/5.3.1
## acml/open64/mp/fma4/5.3.1
## acml/open64-int64/64/5.3.1
## acml/open64-int64/fma4/5.3.1
## acml/open64-int64/mp/64/5.3.1
## acml/open64-int64/mp/fma4/5.3.1
## aspera/AsperaConnect
## bbc/10x_bamtofastq/bamtofastq-1.2.0
## bbc/10x_bamtofastq/bamtofastq-1.3.2
## bbc/10x_subset-bam/10x_subset-bam-1.0
## bbc/7zip/7zip-16.02
## bbc/abyss/abyss-2.2.3
## bbc/AWS/aws-cli
## bbc/bamtools/bamtools-2.5.1
## bbc/bbc_projects/bbc_projects
## bbc/bcftools/bcftools-1.10.2
## bbc/bcftools/bcftools-1.12
## bbc/bcl2fastq/fastq-multx
## bbc/bcl2fastq2/bcl2fastq2-2.20.0
## bbc/bedops/bedops-2.4.37
## bbc/bedtools/bedtools-2.29.2
## bbc/bedtools/bedtools-2.30.0
## bbc/bioawk/bioawk-git
## bbc/biscuit/biscuit_0_3_14
## bbc/biscuit/biscuit_0_3_16
## bbc/biscuit/biscuit_1_0_0
## bbc/biscuit/biscuit_1_0_1
## bbc/biscuit/biscuit_dev
## bbc/bismark/bismark-0.22.3
## bbc/bismark/bismark-0.23.0
## bbc/blast+/blast+-2.10.0
## bbc/bonito/bonito-0.0.5
## bbc/bowtie/bowtie-1.2.3</code></pre>
<p>Print the first 50 lines after subsetting to just the BBC-installed modules.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb17-1"><a href="a-toy-bioinformatics-project.html#cb17-1"></a><span class="ex">module</span> av bbc <span class="op">2&gt;&amp;1</span> <span class="kw">|</span> <span class="fu">head</span> -n50</span></code></pre></div>
<pre><code>## 
## ---------------------------- /cm/shared/modulefiles ----------------------------
## bbc/10x_bamtofastq/bamtofastq-1.2.0
## bbc/10x_bamtofastq/bamtofastq-1.3.2
## bbc/10x_subset-bam/10x_subset-bam-1.0
## bbc/7zip/7zip-16.02
## bbc/abyss/abyss-2.2.3
## bbc/AWS/aws-cli
## bbc/bamtools/bamtools-2.5.1
## bbc/bbc_projects/bbc_projects
## bbc/bcftools/bcftools-1.10.2
## bbc/bcftools/bcftools-1.12
## bbc/bcl2fastq/fastq-multx
## bbc/bcl2fastq2/bcl2fastq2-2.20.0
## bbc/bedops/bedops-2.4.37
## bbc/bedtools/bedtools-2.29.2
## bbc/bedtools/bedtools-2.30.0
## bbc/bioawk/bioawk-git
## bbc/biscuit/biscuit_0_3_14
## bbc/biscuit/biscuit_0_3_16
## bbc/biscuit/biscuit_1_0_0
## bbc/biscuit/biscuit_1_0_1
## bbc/biscuit/biscuit_dev
## bbc/bismark/bismark-0.22.3
## bbc/bismark/bismark-0.23.0
## bbc/blast+/blast+-2.10.0
## bbc/bonito/bonito-0.0.5
## bbc/bowtie/bowtie-1.2.3
## bbc/bowtie2/bowtie2-2.3.5.1
## bbc/bowtie2/bowtie2-2.4.1
## bbc/brename/brename-v2.11.1
## bbc/bustools/bustools-0.39.3
## bbc/bustools/bustools-0.40.0
## bbc/bwa/bwa-0.7.17
## bbc/bwa-mem2/bwa-mem2-v2.2.1
## bbc/byacc/byacc-1.9
## bbc/canu/canu-1.9
## bbc/CaVEMan/CaVEMan-1.15.1
## bbc/cellranger/cellranger-3.0.2
## bbc/cellranger/cellranger-3.1.0
## bbc/cellranger/cellranger-4.0.0
## bbc/cellranger/cellranger-6.0.2
## bbc/cellranger/cellranger-6.1.2
## bbc/cellranger-atac/cellranger-atac-1.1.0
## bbc/changeo/changeo-1.1.0
## bbc/choose/choose-1.3.3
## bbc/chromap/chromap
## bbc/ChromHMM/chromHMM-1.22
## bbc/ChromHMM/chromHMM-1.23
## bbc/cmake/cmake-3.19.4</code></pre>
<p>Print the modules with the keyword, ‘fastqc’, in their names.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="a-toy-bioinformatics-project.html#cb19-1"></a><span class="ex">module</span> av <span class="op">2&gt;&amp;1</span> <span class="kw">|</span> <span class="fu">grep</span> <span class="st">&#39;fastqc&#39;</span></span></code></pre></div>
<pre><code>## bbc/fastqc/fastqc-0.11.8
## bbc/fastqc/fastqc-0.11.9</code></pre>
</div>
<div id="run-fastqc-on-the-fastq-files" class="section level2 hasAnchor">
<h2><span class="header-section-number">4.8</span> Run FastQC on the fastq files<a href="a-toy-bioinformatics-project.html#run-fastqc-on-the-fastq-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We run fastqc on our raw fastq files for essentially all of the projects that we work on.</p>
<p>The output of fastqc includes an html file for each fastq file. You can open up these in a regular browser to look at basic QC metrics for these fastq files. In the interest of time, we will simply run FastQC without viewing the results because we will aggregate all of our results into one html file at the end of this exercise.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="a-toy-bioinformatics-project.html#cb21-1"></a><span class="ex">module</span> load bbc/fastqc/fastqc-0.11.9</span>
<span id="cb21-2"><a href="a-toy-bioinformatics-project.html#cb21-2"></a></span>
<span id="cb21-3"><a href="a-toy-bioinformatics-project.html#cb21-3"></a><span class="co"># make a directory for the output</span></span>
<span id="cb21-4"><a href="a-toy-bioinformatics-project.html#cb21-4"></a><span class="fu">mkdir</span> fastqc</span>
<span id="cb21-5"><a href="a-toy-bioinformatics-project.html#cb21-5"></a></span>
<span id="cb21-6"><a href="a-toy-bioinformatics-project.html#cb21-6"></a><span class="co"># run FastQC. Remember you can check out what the options do by typing `fastqc -h`.</span></span>
<span id="cb21-7"><a href="a-toy-bioinformatics-project.html#cb21-7"></a><span class="ex">fastqc</span> -o fastqc fastqs/*fastq.gz</span></code></pre></div>
<pre><code>## Started analysis of SRR1039520_1.fastq.gz
## Approx 5% complete for SRR1039520_1.fastq.gz
## Approx 10% complete for SRR1039520_1.fastq.gz
## Approx 15% complete for SRR1039520_1.fastq.gz
## Approx 20% complete for SRR1039520_1.fastq.gz
## Approx 25% complete for SRR1039520_1.fastq.gz
## Approx 30% complete for SRR1039520_1.fastq.gz
## Approx 35% complete for SRR1039520_1.fastq.gz
## Approx 40% complete for SRR1039520_1.fastq.gz
## Approx 45% complete for SRR1039520_1.fastq.gz
## Approx 50% complete for SRR1039520_1.fastq.gz
## Approx 55% complete for SRR1039520_1.fastq.gz
## Approx 60% complete for SRR1039520_1.fastq.gz
## Approx 65% complete for SRR1039520_1.fastq.gz
## Approx 70% complete for SRR1039520_1.fastq.gz
## Approx 75% complete for SRR1039520_1.fastq.gz
## Approx 80% complete for SRR1039520_1.fastq.gz
## Approx 85% complete for SRR1039520_1.fastq.gz
## Approx 90% complete for SRR1039520_1.fastq.gz
## Approx 95% complete for SRR1039520_1.fastq.gz
## Approx 100% complete for SRR1039520_1.fastq.gz
## Analysis complete for SRR1039520_1.fastq.gz
## Started analysis of SRR1039520_2.fastq.gz
## Approx 5% complete for SRR1039520_2.fastq.gz
## Approx 10% complete for SRR1039520_2.fastq.gz
## Approx 15% complete for SRR1039520_2.fastq.gz
## Approx 20% complete for SRR1039520_2.fastq.gz
## Approx 25% complete for SRR1039520_2.fastq.gz
## Approx 30% complete for SRR1039520_2.fastq.gz
## Approx 35% complete for SRR1039520_2.fastq.gz
## Approx 40% complete for SRR1039520_2.fastq.gz
## Approx 45% complete for SRR1039520_2.fastq.gz
## Approx 50% complete for SRR1039520_2.fastq.gz
## Approx 55% complete for SRR1039520_2.fastq.gz
## Approx 60% complete for SRR1039520_2.fastq.gz
## Approx 65% complete for SRR1039520_2.fastq.gz
## Approx 70% complete for SRR1039520_2.fastq.gz
## Approx 75% complete for SRR1039520_2.fastq.gz
## Approx 80% complete for SRR1039520_2.fastq.gz
## Approx 85% complete for SRR1039520_2.fastq.gz
## Approx 90% complete for SRR1039520_2.fastq.gz
## Approx 95% complete for SRR1039520_2.fastq.gz
## Approx 100% complete for SRR1039520_2.fastq.gz
## Analysis complete for SRR1039520_2.fastq.gz
## Started analysis of SRR1039521_1.fastq.gz
## Approx 5% complete for SRR1039521_1.fastq.gz
## Approx 10% complete for SRR1039521_1.fastq.gz
## Approx 15% complete for SRR1039521_1.fastq.gz
## Approx 20% complete for SRR1039521_1.fastq.gz
## Approx 25% complete for SRR1039521_1.fastq.gz
## Approx 30% complete for SRR1039521_1.fastq.gz
## Approx 35% complete for SRR1039521_1.fastq.gz
## Approx 40% complete for SRR1039521_1.fastq.gz
## Approx 45% complete for SRR1039521_1.fastq.gz
## Approx 50% complete for SRR1039521_1.fastq.gz
## Approx 55% complete for SRR1039521_1.fastq.gz
## Approx 60% complete for SRR1039521_1.fastq.gz
## Approx 65% complete for SRR1039521_1.fastq.gz
## Approx 70% complete for SRR1039521_1.fastq.gz
## Approx 75% complete for SRR1039521_1.fastq.gz
## Approx 80% complete for SRR1039521_1.fastq.gz
## Approx 85% complete for SRR1039521_1.fastq.gz
## Approx 90% complete for SRR1039521_1.fastq.gz
## Approx 95% complete for SRR1039521_1.fastq.gz
## Approx 100% complete for SRR1039521_1.fastq.gz
## Analysis complete for SRR1039521_1.fastq.gz
## Started analysis of SRR1039521_2.fastq.gz
## Approx 5% complete for SRR1039521_2.fastq.gz
## Approx 10% complete for SRR1039521_2.fastq.gz
## Approx 15% complete for SRR1039521_2.fastq.gz
## Approx 20% complete for SRR1039521_2.fastq.gz
## Approx 25% complete for SRR1039521_2.fastq.gz
## Approx 30% complete for SRR1039521_2.fastq.gz
## Approx 35% complete for SRR1039521_2.fastq.gz
## Approx 40% complete for SRR1039521_2.fastq.gz
## Approx 45% complete for SRR1039521_2.fastq.gz
## Approx 50% complete for SRR1039521_2.fastq.gz
## Approx 55% complete for SRR1039521_2.fastq.gz
## Approx 60% complete for SRR1039521_2.fastq.gz
## Approx 65% complete for SRR1039521_2.fastq.gz
## Approx 70% complete for SRR1039521_2.fastq.gz
## Approx 75% complete for SRR1039521_2.fastq.gz
## Approx 80% complete for SRR1039521_2.fastq.gz
## Approx 85% complete for SRR1039521_2.fastq.gz
## Approx 90% complete for SRR1039521_2.fastq.gz
## Approx 95% complete for SRR1039521_2.fastq.gz
## Approx 100% complete for SRR1039521_2.fastq.gz
## Analysis complete for SRR1039521_2.fastq.gz</code></pre>
<p>Look at the output from FastQC.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="a-toy-bioinformatics-project.html#cb23-1"></a><span class="fu">ls</span> fastqc</span></code></pre></div>
<pre><code>## SRR1039520_1_fastqc.html
## SRR1039520_1_fastqc.zip
## SRR1039520_2_fastqc.html
## SRR1039520_2_fastqc.zip
## SRR1039521_1_fastqc.html
## SRR1039521_1_fastqc.zip
## SRR1039521_2_fastqc.html
## SRR1039521_2_fastqc.zip</code></pre>
</div>
<div id="set-up-a-job-to-run-salmon" class="section level2 hasAnchor">
<h2><span class="header-section-number">4.9</span> Set up a job to run Salmon<a href="a-toy-bioinformatics-project.html#set-up-a-job-to-run-salmon" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First, we <code>exit</code> from our interactive job because we want to get back on to the submit node to submit a non-interactive job to run Salmon.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb25-1"><a href="a-toy-bioinformatics-project.html#cb25-1"></a><span class="co"># exit the interactive job</span></span>
<span id="cb25-2"><a href="a-toy-bioinformatics-project.html#cb25-2"></a><span class="bu">exit</span></span>
<span id="cb25-3"><a href="a-toy-bioinformatics-project.html#cb25-3"></a></span>
<span id="cb25-4"><a href="a-toy-bioinformatics-project.html#cb25-4"></a><span class="co"># verify that the interactive job has ended</span></span>
<span id="cb25-5"><a href="a-toy-bioinformatics-project.html#cb25-5"></a><span class="ex">qstat</span> -u username</span>
<span id="cb25-6"><a href="a-toy-bioinformatics-project.html#cb25-6"></a></span>
<span id="cb25-7"><a href="a-toy-bioinformatics-project.html#cb25-7"></a><span class="co"># go back to the project directory</span></span>
<span id="cb25-8"><a href="a-toy-bioinformatics-project.html#cb25-8"></a><span class="bu">cd</span> /varidata/researchtemp/hpctmp/HPC_mini_workshop/Part3/firstname.lastname</span></code></pre></div>
<p>Next, we can set up our job script to run Salmon. The job script code is below. Don’t worry about the specifics of this code for now. Simply, copy the code and paste into your text editor to make your job script. Save the script as <code>run_salmon.sh</code> and ensure that it is in our project directory containing the <code>fastqs/</code> subfolder.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb26-1"><a href="a-toy-bioinformatics-project.html#cb26-1"></a><span class="co">#PBS -l nodes=1:ppn=3</span></span>
<span id="cb26-2"><a href="a-toy-bioinformatics-project.html#cb26-2"></a><span class="co">#PBS -l mem=36gb</span></span>
<span id="cb26-3"><a href="a-toy-bioinformatics-project.html#cb26-3"></a><span class="co">#PBS -l walltime=1:00:00</span></span>
<span id="cb26-4"><a href="a-toy-bioinformatics-project.html#cb26-4"></a><span class="co">#PBS -N salmon</span></span>
<span id="cb26-5"><a href="a-toy-bioinformatics-project.html#cb26-5"></a><span class="co">#PBS -o salmon.o</span></span>
<span id="cb26-6"><a href="a-toy-bioinformatics-project.html#cb26-6"></a><span class="co">#PBS -e salmon.e</span></span>
<span id="cb26-7"><a href="a-toy-bioinformatics-project.html#cb26-7"></a></span>
<span id="cb26-8"><a href="a-toy-bioinformatics-project.html#cb26-8"></a><span class="kw">set</span> <span class="ex">-euo</span> pipefail <span class="co"># see explanation at https://gist.github.com/mohanpedala/1e2ff5661761d3abd0385e8223e16425?permalink_comment_id=3945021</span></span>
<span id="cb26-9"><a href="a-toy-bioinformatics-project.html#cb26-9"></a></span>
<span id="cb26-10"><a href="a-toy-bioinformatics-project.html#cb26-10"></a><span class="va">start_time=$(</span><span class="fu">date</span> +<span class="st">&quot;%T&quot;</span><span class="va">)</span></span>
<span id="cb26-11"><a href="a-toy-bioinformatics-project.html#cb26-11"></a></span>
<span id="cb26-12"><a href="a-toy-bioinformatics-project.html#cb26-12"></a><span class="co"># You need to navigate to your project directory. Conveniently, the $PBS_O_WORKDIR variable stores the path for where the job was submitted.</span></span>
<span id="cb26-13"><a href="a-toy-bioinformatics-project.html#cb26-13"></a><span class="bu">cd</span> <span class="va">${PBS_O_WORKDIR}</span></span>
<span id="cb26-14"><a href="a-toy-bioinformatics-project.html#cb26-14"></a></span>
<span id="cb26-15"><a href="a-toy-bioinformatics-project.html#cb26-15"></a><span class="ex">module</span> load bbc/salmon/salmon-1.5.2</span>
<span id="cb26-16"><a href="a-toy-bioinformatics-project.html#cb26-16"></a></span>
<span id="cb26-17"><a href="a-toy-bioinformatics-project.html#cb26-17"></a><span class="co"># Typically, you would have to first build an index before doing the aligning, but we have done this for you already. Here, we store the path to the index file in a variable called &#39;salmon_idx&#39;.</span></span>
<span id="cb26-18"><a href="a-toy-bioinformatics-project.html#cb26-18"></a><span class="va">salmon_idx=</span><span class="st">&quot;/varidata/research/projects/bbc/versioned_references/2022-03-08_14.47.50_v9/data/hg38_gencode/indexes/salmon/hg38_gencode/&quot;</span></span>
<span id="cb26-19"><a href="a-toy-bioinformatics-project.html#cb26-19"></a></span>
<span id="cb26-20"><a href="a-toy-bioinformatics-project.html#cb26-20"></a><span class="co"># make output directory for salmon</span></span>
<span id="cb26-21"><a href="a-toy-bioinformatics-project.html#cb26-21"></a><span class="fu">mkdir</span> -p salmon</span>
<span id="cb26-22"><a href="a-toy-bioinformatics-project.html#cb26-22"></a></span>
<span id="cb26-23"><a href="a-toy-bioinformatics-project.html#cb26-23"></a><span class="co"># this is called a for loop. We use this to run salmon quant on all the samples, one at a time. It is more efficient to run salmon on each sample &quot;in parallel&quot; but we will not do not today.</span></span>
<span id="cb26-24"><a href="a-toy-bioinformatics-project.html#cb26-24"></a><span class="kw">for</span> <span class="ex">samp_id</span> in SRR1039520 SRR1039521</span>
<span id="cb26-25"><a href="a-toy-bioinformatics-project.html#cb26-25"></a><span class="kw">do</span></span>
<span id="cb26-26"><a href="a-toy-bioinformatics-project.html#cb26-26"></a>    <span class="ex">salmon</span> quant -p <span class="va">${PBS_NUM_PPN}</span> -l A -i <span class="va">$salmon_idx</span> -1 fastqs/<span class="va">${samp_id}</span>_1.fastq.gz -2 fastqs/<span class="va">${samp_id}</span>_2.fastq.gz -o salmon/<span class="va">${samp_id}</span> --validateMappings</span>
<span id="cb26-27"><a href="a-toy-bioinformatics-project.html#cb26-27"></a></span>
<span id="cb26-28"><a href="a-toy-bioinformatics-project.html#cb26-28"></a><span class="kw">done</span></span>
<span id="cb26-29"><a href="a-toy-bioinformatics-project.html#cb26-29"></a></span>
<span id="cb26-30"><a href="a-toy-bioinformatics-project.html#cb26-30"></a><span class="va">end_time=$(</span><span class="fu">date</span> +<span class="st">&quot;%T&quot;</span><span class="va">)</span></span>
<span id="cb26-31"><a href="a-toy-bioinformatics-project.html#cb26-31"></a><span class="bu">echo</span> <span class="st">&quot;Start time: </span><span class="va">$start_time</span><span class="st">&quot;</span></span>
<span id="cb26-32"><a href="a-toy-bioinformatics-project.html#cb26-32"></a><span class="bu">echo</span> <span class="st">&quot;End time: </span><span class="va">$end_time</span><span class="st">&quot;</span></span></code></pre></div>
</div>
<div id="submit-the-job" class="section level2 hasAnchor">
<h2><span class="header-section-number">4.10</span> Submit the job<a href="a-toy-bioinformatics-project.html#submit-the-job" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Submit the job and keep monitoring the job using <code>qstat</code> as shown below. The job should complete in about 2 minutes.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb27-1"><a href="a-toy-bioinformatics-project.html#cb27-1"></a><span class="ex">qsub</span> run_salmon.sh</span>
<span id="cb27-2"><a href="a-toy-bioinformatics-project.html#cb27-2"></a></span>
<span id="cb27-3"><a href="a-toy-bioinformatics-project.html#cb27-3"></a><span class="co"># check on the status of your job</span></span>
<span id="cb27-4"><a href="a-toy-bioinformatics-project.html#cb27-4"></a><span class="ex">qstat</span> -u firstname.lastname</span></code></pre></div>
</div>
<div id="check-the-job-logs-to-see-if-job-finished-running" class="section level2 hasAnchor">
<h2><span class="header-section-number">4.11</span> Check the job logs to see if job finished running<a href="a-toy-bioinformatics-project.html#check-the-job-logs-to-see-if-job-finished-running" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>It is good practice to check the job logs after your job is done to ensure that the job completed successfully. If there is an error, the output files may not be reliable or could be incomplete.</p>
<p>Look at the stderr from the script.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb28-1"><a href="a-toy-bioinformatics-project.html#cb28-1"></a><span class="fu">tail</span> salmon.e</span></code></pre></div>
<pre><code>## [2022-09-20 23:32:16.397] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate
## [2022-09-20 23:32:16.426] [jointLog] [info] iteration = 0 | max rel diff. = 202.929
## [2022-09-20 23:32:18.786] [jointLog] [info] iteration = 100 | max rel diff. = 6.3715
## [2022-09-20 23:32:21.080] [jointLog] [info] iteration = 200 | max rel diff. = 5.04421
## [2022-09-20 23:32:23.395] [jointLog] [info] iteration = 300 | max rel diff. = 0.631388
## [2022-09-20 23:32:25.725] [jointLog] [info] iteration = 400 | max rel diff. = 2.25551
## [2022-09-20 23:32:27.268] [jointLog] [info] iteration = 467 | max rel diff. = 0.00956146
## [2022-09-20 23:32:27.307] [jointLog] [info] Finished optimizer
## [2022-09-20 23:32:27.307] [jointLog] [info] writing output</code></pre>
<p>Look at the stdout from the script.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb30-1"><a href="a-toy-bioinformatics-project.html#cb30-1"></a><span class="fu">tail</span> salmon.o</span></code></pre></div>
<pre><code>## Start time: 23:30:31
## End time: 23:32:30</code></pre>
</div>
<div id="use-grep-to-find-the-tpms-for-specific-genes" class="section level2 hasAnchor">
<h2><span class="header-section-number">4.12</span> Use grep to find the TPMs for specific genes<a href="a-toy-bioinformatics-project.html#use-grep-to-find-the-tpms-for-specific-genes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As an example, let’s try to extract out the TPMs for MUC1. These values can be found in the <code>quant.sf</code> file in each sample’s folder. Let’s take a look at one of these files to figure out the format of these files.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb32-1"><a href="a-toy-bioinformatics-project.html#cb32-1"></a><span class="fu">head</span> salmon/SRR1039520/quant.sf </span></code></pre></div>
<pre><code>## Name Length  EffectiveLength TPM NumReads
## ENST00000456328.2    1657    1502.015    0.000000    0.000
## ENST00000450305.2    632 477.136 0.000000    0.000
## ENST00000488147.1    1351    1196.015    0.000000    0.000
## ENST00000619216.1    68  1.932   0.000000    0.000
## ENST00000473358.1    712 557.094 0.000000    0.000
## ENST00000469289.1    535 380.252 0.000000    0.000
## ENST00000607096.1    138 24.897  0.000000    0.000
## ENST00000417324.1    1187    1032.015    0.000000    0.000
## ENST00000461467.1    590 435.180 0.000000    0.000</code></pre>
<p>We can see that the TPMs are in the 4th column of this file. We can also see that the transcript IDs are in ENSEMBL format. If we look up MUC1 on <a href="http://useast.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000185499;r=1:155185824-155192916">Ensembl</a>, we will find that the canonical transcript is <code>ENST00000620103</code>, so let’s <code>grep</code> for this transcript ID.</p>
<p>Look for this transript for one specific sample.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb34-1"><a href="a-toy-bioinformatics-project.html#cb34-1"></a><span class="fu">grep</span> <span class="st">&#39;ENST00000620103&#39;</span> salmon/SRR1039520/quant.sf</span></code></pre></div>
<pre><code>## ENST00000620103.4    1811    1656.015    0.000000    0.000</code></pre>
<p>Look across all the samples at the same time. We can see that ‘SRR1039521’ has a TPM of 13.1 for MUC1 compared to 0 for ‘SRR1039520’. Recall that the fastq files for this exercise were subsetted to a very small number of reads so don’t interpret these results seriously.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb36-1"><a href="a-toy-bioinformatics-project.html#cb36-1"></a><span class="fu">grep</span> <span class="st">&#39;ENST00000620103&#39;</span> salmon/*/quant.sf</span></code></pre></div>
<pre><code>## salmon/SRR1039520/quant.sf:ENST00000620103.4 1811    1656.015    0.000000    0.000
## salmon/SRR1039521/quant.sf:ENST00000620103.4 1811    1655.481    13.145898   5.723</code></pre>
</div>
<div id="use-an-interactive-job-to-run-multiqc-on-the-salmon-and-fastqc-output" class="section level2 hasAnchor">
<h2><span class="header-section-number">4.13</span> Use an interactive job to run multiQC on the Salmon and FastQC output<a href="a-toy-bioinformatics-project.html#use-an-interactive-job-to-run-multiqc-on-the-salmon-and-fastqc-output" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As our final exercise, let’s run a tool called multiQC to summarize the results from FastQC and Salmon. Using the output from multiQC, we will be able to assess both the quality of the fastq sequences themselves and how well they match up with the reference transcriptome (mapping rate), and other useful information. We will start up another interactive job to do this.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb38-1"><a href="a-toy-bioinformatics-project.html#cb38-1"></a><span class="ex">qsub</span> -I -l nodes=1:ppn=1 -l mem=32gb -l walltime=1:00:00</span>
<span id="cb38-2"><a href="a-toy-bioinformatics-project.html#cb38-2"></a></span>
<span id="cb38-3"><a href="a-toy-bioinformatics-project.html#cb38-3"></a><span class="co"># go back to your project directory</span></span>
<span id="cb38-4"><a href="a-toy-bioinformatics-project.html#cb38-4"></a><span class="bu">cd</span> /varidata/researchtemp/hpctmp/HPC_mini_workshop/Part3/firstname.lastname/</span></code></pre></div>
<p>Now actually run multiQC.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb39-1"><a href="a-toy-bioinformatics-project.html#cb39-1"></a><span class="ex">module</span> load bbc/multiqc/multiqc-1.12</span>
<span id="cb39-2"><a href="a-toy-bioinformatics-project.html#cb39-2"></a></span>
<span id="cb39-3"><a href="a-toy-bioinformatics-project.html#cb39-3"></a><span class="co"># multiqc creates the output directory automatically. You don&#39;t have to run mkdir manually.</span></span>
<span id="cb39-4"><a href="a-toy-bioinformatics-project.html#cb39-4"></a><span class="ex">multiqc</span> --outdir multiqc .</span></code></pre></div>
<pre><code>## 
## ### Loaded BBC module
##  Loading this module prepends to $PYTHONPATH
##  Don&#39;t use with conda.
## ### End BBC module message.
## 
##   /// MultiQC 🔍 | v1.12
## 
## |           multiqc | Search path : /varidata/research/home/daisy.fu/Intro_to_Linux_for_HPC
## |         searching | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 106/106  
## |            salmon | Found 2 meta reports
## |            salmon | Found 2 fragment length distributions
## |            fastqc | Found 4 reports
## |           multiqc | Compressing plot data
## |           multiqc | Report      : multiqc/multiqc_report.html
## |           multiqc | Data        : multiqc/multiqc_data
## |           multiqc | MultiQC complete</code></pre>
<p>Look at the output from multiQC.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb41-1"><a href="a-toy-bioinformatics-project.html#cb41-1"></a><span class="fu">ls</span> multiqc</span></code></pre></div>
<pre><code>## multiqc_data
## multiqc_report.html</code></pre>
<p>Note the newly created <code>multiqc_report.html</code> file. Your final task today is to view this file in your browser. If you have mounted the HPC file system to your computer, you can try to open up this file directly. Alternatively, you can copy this file to your computer’s local storage first and then open it.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="hpc-basics-and-the-job-scheduler.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="assets/gitbook-2.6.7/js/app.min.js"></script>
<script src="assets/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="assets/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
},
"toolbar": {
"position": "static"
}
});
});
</script>

</body>

</html>
